app/components/new-directory-user.js

```diff
@@ -1,127 +1,208 @@
-          return allowCustomUserName;
-        });
-      }),
-    }),
-  ],
-  otherId: [
-    validator('length', {
-      max: 16,
-    }),
-  ],
-});
-
-export default Component.extend(NewUser, Validations, {
-  fetch: service(),
-  iliosConfig: service(),
-  intl: service(),
-
-  classNames: ['new-directory-user'],
-
-  isSearching: false,
-  searchResults: null,
-  searchResultsReturned: false,
-  searchTerms: null,
-  selectedUser: false,
-  tooManyResults: false,
-
-  allowCustomUserName: computed('iliosConfig.authenticationType', async function () {
-    const type = await this.iliosConfig.authenticationType;
-    return type === 'form';
-  }),
-
-  init() {
-    this._super(...arguments);
-    this.set('searchResults', []);
-    const searchTerms = this.searchTerms;
-    if (isPresent(searchTerms)) {
-      this.findUsersInDirectory.perform(searchTerms);
+import { all, filter } from 'rsvp';
+import { dropTask, restartableTask } from 'ember-concurrency';
+import moment from 'moment';
+import { validatable, Length, NotBlank } from 'ilios-common/decorators/validation';
+import { ValidateIf } from 'class-validator';
+
+@validatable
+export default class NewDirectoryUserComponent extends Component {
+  @service fetch;
+  @service iliosConfig;
+  @service intl;
+  @service store;
+  @service currentUser;
+  @service flashMessages;
+  @service permissionChecker;
+
+  @tracked allowCustomUserName = false;
+  @tracked isSearching = false;
+  @tracked searchResults = [];
+  @tracked searchResultsReturned = false;
+  @tracked selectedUser = false;
+  @tracked tooManyResults = false;
+  @tracked firstName;
+  @tracked middleName;
+  @tracked lastName;
+  @tracked campusId;
+  @tracked @Length(0, 16) otherId;
+  @tracked email;
+  @tracked @ValidateIf((o) => o.allowCustomUserName) @NotBlank() @Length(1, 100) username;
+  @tracked @ValidateIf((o) => o.allowCustomUserName) @NotBlank() password;
+  @tracked phone;
+  @tracked schoolId = null;
+  @tracked primaryCohortId = null;
+  @tracked primarySchool = null;
+  @tracked currentSchoolCohorts = [];
+  @tracked cohorts = [];
+  @tracked schools = [];
+  @tracked isSaving = false;
+  @tracked nonStudentMode = true;
+
+  get isLoading() {
+    return this.load.isRunning || this.reload.isRunning;
+  }
+
+  get bestSelectedSchool() {
+    if (this.schoolId) {
+      const currentSchool = this.schools.findBy('id', this.schoolId);
+
+      if (currentSchool) {
+        return currentSchool;
+      }
+    }
+    return this.primarySchool;
+  }
+
+  get bestSelectedCohort() {
+    if (!this.currentSchoolCohorts) {
+      return null;
+    }
+
+    if (this.primaryCohortId) {
+      const currentCohort = this.currentSchoolCohorts.findBy('id', this.primaryCohortId);
+
+      if (currentCohort) {
+        return currentCohort;
+      }
+    }
+
+    return this.currentSchoolCohorts.lastObject;
+  }
+
+  async loadSchools() {
+    const schools = await this.store.findAll('school', { reload: true });
```

Comment:
I don't think the `reload: true` is necessary anymore. This seems like a workaround we may have needed in the past.

Reply:
:heavy_check_mark: